src/pages/patch/PatchRestartModal.tsx

```diff
@@ -92,26 +201,108 @@ export const PatchRestartModal: React.FC<PatchModalProps> = ({
   }, [search, prevSearch, setBaseStatusFilterTerm, setPatchStatusFilterTerm]);
 
   const patchAnalytics = usePatchAnalytics();
-  const handlePatchRestart = async (e): Promise<void> => {
-    console.log(childPatches);
-    e.preventDefault();
-    try {
-      patchAnalytics.sendEvent({
-        name: "Restart",
-        abort: shouldAbortInProgressTasks,
-      });
+  const handlePatchRestart = async () => {
+    patchAnalytics.sendEvent({
+      name: "Restart",
+      abort: shouldAbortInProgressTasks,
+    });
+    if (selectedArray(selectedTasks).length !== 0) {
       await restartPatch({
         variables: {
           patchId,
           taskIds: selectedArray(selectedTasks),
           abort: shouldAbortInProgressTasks,
         },
       });
-    } catch {
-      // This is handled in the onError handler for the mutation
     }
   };
 
+  if (restart !== prevRestart) {
+    handlePatchRestart();
+  }
+
+  return (
+    <>
+      {patchBuildVariants && (
+        <>
+          <Row>
+            <TaskStatusFilters
+              onChangeBaseStatusFilter={setBaseStatusFilterTerm}
+              onChangeStatusFilter={setPatchStatusFilterTerm}
+              patchId={patchId}
+              selectedBaseStatuses={baseStatusFilterTerm}
+              selectedStatuses={patchStatusFilterTerm}
+              filterWidth="50%"
+            />
+          </Row>
+          {patchBuildVariants.map((patchBuildVariant) => (
+            <PatchBuildVariantAccordion
+              key={`accoridan_${patchBuildVariant.variant}`}
+              tasks={patchBuildVariant.tasks}
+              displayName={patchBuildVariant.displayName}
+              selectedTasks={selectedTasks}
+              toggleSelectedTask={toggleSelectedTask}
+            />
+          ))}
+        </>
+      )}
+    </>
+  );
+};
+
+interface PatchRestartFooterProps {
+  setShouldAbortInProgressTasks: React.Dispatch<React.SetStateAction<boolean>>;
+  shouldAbortInProgressTasks: boolean;
+}
+export const PatchRestartFooter: React.FC<PatchRestartFooterProps> = ({
+  setShouldAbortInProgressTasks,
+  shouldAbortInProgressTasks,
+}) => (
+  <>
+    <HR />
+    <ConfirmationMessage weight="medium" data-cy="confirmation-message">
+      Are you sure you want to restart the selected tasks?
+    </ConfirmationMessage>
+    <Checkbox
+      onChange={() =>
+        setShouldAbortInProgressTasks(!shouldAbortInProgressTasks)
+      }
+      label="Abort in progress tasks"
+      checked={shouldAbortInProgressTasks}
+      bold={false}
+    />
+  </>
+);
+
+interface PatchModalProps {
+  visible: boolean;
+  onOk: () => void;
+  onCancel: () => void;
+  patchId?: string;
+  refetchQueries: string[];
+  childPatches: Partial<Patch>[];
+}
+export const PatchRestartModal: React.FC<PatchModalProps> = ({
```

Comment:
can we move this into a separate file where the export name matches the filename